File: Comms Protocol Java/Comms Protocol Java/comms/protocol/java/Rotation_Vector_RawDataPacket.java
package comms.protocol.java;

public class Rotation_Vector_RawDataPacket extends AndroidDataPacket {
	private static int numberOfValues = 4;

	public Rotation_Vector_RawDataPacket() {
		super(ValidPacketTypes.rotation_vector_raw_data_packet, numberOfValues);
	}

	public Rotation_Vector_RawDataPacket(byte[] payload) throws Exception {
		super(ValidPacketTypes.rotation_vector_raw_data_packet, numberOfValues, payload);
	}

	public Rotation_Vector_RawDataPacket(DataPacket packet) throws Exception {
		super(ValidPacketTypes.rotation_vector_raw_data_packet, numberOfValues, packet);
	}

	public Quaternion GetQuat() {
		AndroidSensor sensor = DeSerialize();
		return new Quaternion(sensor.GetValues());
	}

}

File: Comms Protocol CSharp/Comms Protocol CSharp/RotationVectorRawDataPacket.cs

namespace Comms_Protocol_CSharp
{
    public class RotationVectorRawDataPacket : AndroidDataPacket
    {
        private static int numberOfVals = 4;

        public RotationVectorRawDataPacket() : base(ValidPacketTypes.rotation_vector_raw_data_packet, numberOfVals)
        {

        }

        public RotationVectorRawDataPacket(byte[] payload) : base(ValidPacketTypes.rotation_vector_raw_data_packet, numberOfVals, payload)
        {

        }

        public RotationVectorRawDataPacket(DataPacket packet) : base(ValidPacketTypes.rotation_vector_raw_data_packet, numberOfVals, packet)
        {

        }

        public Quaternion GetQuat()
        {
            AndroidSensor sensor = DeSerialize();
            return new Quaternion(sensor.GetValues());
        }
    }
}
